ee/src/main/java/org/jboss/as/ee/subsystem/ContextServiceAdd.java

```diff
@@ -48,15 +47,16 @@ private ContextServiceAdd() {
 
     @Override
     protected void performRuntime(OperationContext context, ModelNode operation, Resource resource) throws OperationFailedException {
-        ModelNode model = resource.getModel();
-        final String name = PathAddress.pathAddress(operation.get(ModelDescriptionConstants.ADDRESS)).getLastElement().getValue();
+        final ModelNode model = resource.getModel();
+        final String name = context.getCurrentAddressValue();
 
         final String jndiName = ContextServiceResourceDefinition.JNDI_NAME_AD.resolveModelAttribute(context, model).asString();
         final boolean useTransactionSetupProvider = ContextServiceResourceDefinition.USE_TRANSACTION_SETUP_PROVIDER_AD.resolveModelAttribute(context, model).asBoolean();
 
         // install the service which manages the default context service
+        final ServiceName serviceName = ContextServiceResourceDefinition.CAPABILITY.getCapabilityServiceName(name);
         final ContextServiceService contextServiceService = new ContextServiceService(name, jndiName, new DefaultContextSetupProviderImpl());
-        final ServiceBuilder<ContextServiceImpl> serviceBuilder = context.getServiceTarget().addService(ConcurrentServiceNames.getContextServiceServiceName(name), contextServiceService);
+        final ServiceBuilder<ContextServiceImpl> serviceBuilder = context.getServiceTarget().addService(serviceName, contextServiceService);
```

Comment:
Use context.getCapabilityServiceTarget().addCapability(...) if possible.

Reply:
now using getCapabilityServiceTarget(), wrt addCapability(...) seems only possible for not dynamic caps thus sticking with addService(ServiceName)

Reply:
done